Makefile

```diff
@@ -0,0 +1,62 @@
+# This Makefile captures common developer build and test use cases.
+
+MAKEFLAGS += --no-print-directory
+
+# - incrementally compile and install the c++ code
+# - copy the shared objects on top of the python dev build
+.PHONY: update
+update:
+	@cd build && make -j && make install-libtiledbsoma
+	cp dist/lib/* apis/python/src/tiledbsoma/
+
+# - run the clean rule
+# - build the c++ code using the environment's cmake (not the python build-system cmake)
+#   - builds a Release with Debug Info build
+# - install the python in editable mode (dev build)
+.PHONY: install
+install: clean
+	./scripts/bld
+	pip install -v -e apis/python
+
+# - run the clean rule
+# - build the c++ code using the environment's cmake (not the python build-system cmake)
+#   - builds a Debug build
+# - install the python in editable mode (dev build)
+.PHONY: install-debug
+install-debug: clean
+	BUILD_TYPE=Debug ./scripts/bld
+	pip install -v -e apis/python
+
+# - run the clean rule
+# - build the c++ static library for r
+.PHONY: r-build
+r-build: clean
+	R_BUILD=1 ./scripts/bld
+
+# - run c++ unit tests
+# - run pytests
+.PHONY: test
+test: test/soco
+	ctest --test-dir build/libtiledbsoma -C Release --verbose
+	pytest
+
+# install the test data
+test/soco:
+	./apis/python/tools/ingestor \
+		--ifexists replace \
+		--soco \
+		-o test/soco \
+		-n \
+		data/pbmc3k_processed.h5ad \
+		data/10x-pbmc-multiome-v1.0/subset_100_100.h5ad
+
+# - remove the c++ build and dist, and the test data
+.PHONY: clean
+clean:
+	rm -rf build dist test/soco
```

Comment:
Removing `test/soco` here may not be a good idea. I have treated this as a 'constant' and relied on it for testing.

Following some other idioms we could add `distclean` to really wipe things away (including this) but make the test data persist.

Reply:
Good point. The goal was to force recreating the test data when moving to newer code, in case the format changed.

A better approach may be to always write new test data when running `make install`.